activation/atxdb.go

```diff
@@ -201,129 +205,110 @@ func (db *DB) createTraversalActiveSetCounterFunc(countedAtxs map[string]types.A
-
-					penalties[atx.NodeID.Key] = struct{}{} // mark node in penalty
-					delete(countedAtxs, atx.NodeID.Key)    // remove the penalized node from counted
-				}
+					log.FieldNamed("requested_epoch", targetEpoch))
 				continue
 			}
 
-			countedAtxs[atx.NodeID.Key] = id
+			minerWeight[atx.NodeID.Key] = atx.GetWeight()
 		}
 
 		return false, nil
 	}
-
-	return traversalFunc
 }
 
-// CalcActiveSetSize - returns the active set size that matches the view of the contextually valid blocks in the provided layer
-func (db *DB) CalcActiveSetSize(epoch types.EpochID, blocks map[types.BlockID]struct{}) (map[string]struct{}, error) {
+// GetMinerWeightsInEpochFromView returns a map of miner IDs and each one's weight targeting targetEpoch, by traversing
+// the provided view.
+func (db *DB) GetMinerWeightsInEpochFromView(targetEpoch types.EpochID, view map[types.BlockID]struct{}) (map[string]uint64, error) {
 
-	if epoch == 0 {
-		return nil, errors.New("tried to retrieve active set for epoch 0")
+	if targetEpoch == 0 {
+		return nil, errors.New("tried to retrieve miner weights for targetEpoch 0")
 	}
 
-	firstLayerOfPrevEpoch := (epoch - 1).FirstLayer()
+	firstLayerOfPrevEpoch := (targetEpoch - 1).FirstLayer()
 
-	countedAtxs := make(map[string]types.ATXID)
-	penalties := make(map[string]struct{})
+	minerWeight := make(map[string]uint64)
 
-	traversalFunc := db.createTraversalActiveSetCounterFunc(countedAtxs, penalties, db.LayersPerEpoch, epoch)
+	traversalFunc := db.createTraversalFuncForMinerWeights(minerWeight, targetEpoch)
 
 	startTime := time.Now()
-	err := db.meshDb.ForBlockInView(blocks, firstLayerOfPrevEpoch, traversalFunc)
+	err := db.meshDb.ForBlockInView(view, firstLayerOfPrevEpoch, traversalFunc)
 	if err != nil {
 		return nil, err
 	}
-	db.log.With().Info("done calculating active set size",
-		log.Int("size", len(countedAtxs)),
+	db.log.With().Info("done calculating miner weights",
+		log.Int("numMiners", len(minerWeight)),
 		log.String("duration", time.Now().Sub(startTime).String()))
 
-	result := make(map[string]struct{}, len(countedAtxs))
-	for k := range countedAtxs {
-		result[k] = struct{}{}
-	}
-
-	return result, nil
+	return minerWeight, nil
 }
 
-// CalcActiveSetFromView traverses the view found in a - the activation tx and counts number of active ids published
-// in the epoch prior to the epoch that a was published at, this number is the number of active ids in the next epoch
-// the function returns error if the view is not found
-func (db *DB) CalcActiveSetFromView(view []types.BlockID, pubEpoch types.EpochID) (uint32, error) {
-	if pubEpoch < 1 {
-		return 0, fmt.Errorf("publication epoch cannot be less than 1, found %v", pubEpoch)
+// CalcTotalWeightFromView traverses the provided view and returns the total weight of the ATXs found targeting
+// targetEpoch. The function returns error if the view is not found.
+func (db *DB) CalcTotalWeightFromView(view []types.BlockID, targetEpoch types.EpochID) (uint64, error) {
```

Comment:
Do we need to traverse a view to calc miner weight? currently we do not include a view in ATX and just declare the list of ATXs as an active set... where does this traversing happens?